packages/@aws-cdk/integ-tests-alpha/lib/assertions/providers/lambda-handler/http.ts

```diff
@@ -22,8 +24,17 @@ export class HttpHandler extends CustomResourceHandler<HttpRequest, HttpResponse
     } catch (e) {
       // Okay
     }
-    return {
-      apiCallResponse: result,
-    };
+
+    let resp: HttpResponseWrapper | { [key: string]: unknown };
+    if (request.flattenResponse === 'true') {
+      // Flatten and explode JSON fields
+      resp = flatten(deepParseJson({ apiCallResponse: result }));
+    } else {
+      // Otherwise just return the response as-is, without exploding JSON fields
+      resp = { apiCallResponse: result };
+    }
+    console.log(`Returning result ${JSON.stringify(resp)}`);
```

Comment:
It might be worth considering a refactor of the `flattenResponse` in the common `CustomResourceHandler` props, given the SDK handler does pretty much the same thing:

https://github.com/aws/aws-cdk/blob/a2e8dc5e3cff91e64e5062a8fed630fb46982dbf/packages/%40aws-cdk/integ-tests-alpha/lib/assertions/providers/lambda-handler/sdk.ts#L18-L22